Add a unit test for join method coverage

joshu4lim opened the pull request:
[O] Wrote test for feature
[ ] Added changes to CHANGELOG.md
[ ] Bumped version number (delete if unneeded)

**Changes proposed:**
Hello, I'm a student and a first-time contributor. 
One of the coverage missing in 'tables.py' is in the '_join' internal method, in lines [2039-2040](https://github.com/data-8/datascience/blob/master/datascience/tables.py#L2039-L2040) where it checks for an empty table. However, the 'join' method for external interface already has these checks in lines [2024-2025](https://github.com/data-8/datascience/blob/master/datascience/tables.py#L2024-L2025), preventing those lines from being covered. This makes it difficult to test '_join' by just calling 'join' as discussed in this [PR](https://github.com/data-8/datascience/pull/615). 

In this unit test, I decided to modify the 'join' method to skip those checks and allow '_join' to be covered. This way, the functionality of _join can be tested without directly invoking it.

Please let me know if any changes are needed to improve this implementation.

adnanhemani commented:
I'm not really in favor of testing in such roundabout ways. If we have dead code, I would much rather remove it instead.

adnanhemani reviewed (changes requested):
I'm not really in favor of testing in such roundabout ways. If we have dead code, I would much rather remove it instead.

joshu4lim commented:
Removed the proposed roundabout test case and removed the dead code inside the _join internal method on lines [2039-2040](https://github.com/data-8/datascience/blob/master/datascience/tables.py#L2039-L2040). Also removed lines [2041-2042](https://github.com/data-8/datascience/blob/master/datascience/tables.py#L2041-L2042) because it would also be dead code based on the same reasoning for removing L2039-2040.

davidwagner commented:
@adnanhemani I'll let you comment, but it looks reasonable to me.

adnanhemani reviewed (approved):
LGTM - thanks for this change!
